ndb_api/operation_helpers.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2022-2023 Nutanix, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ndb_api
+
+// Returns a boolean indicating if an operation has reached a terminal state
+func HasOperationFailed(o OperationResponse) bool {
+	status := o.Status
+	return status == "2" || status == "3" || status == "4"
+}
+
+// Returns a boolean indicating if an operation was successful
+func HasOperationPassed(o OperationResponse) bool {
```

Comment:
Can just have one function, say getOperationStatus that returns an enum PASS (0) for FAIL(1)?

Reply:
Yes we can, but I think this way I wouldn't have to delegate the status checking in the client code of this package. 
Ultimately, we're still checking for a `PASS` or a `FAIL` state where we decide our next steps/actions which is accomplished with less verbosity if we have these methods. Otherwise, the client of this package might get way too verbose. 
Also, if the status values change in the NDB API (highly unlikely but let us assume), then I might have to make change in only one function and not in all the places this function is called.

Please let me know if you'd want to discuss more on this.